### PR TITLE
(PUP-5545) rubocop by default

### PR DESCRIPTION
**Pull request by MikaelSmith**

### CODE REVIEW

**Comment by andersonmills** [comment]
We got an error.

```
$ rake rubocop
Running RuboCop...
Error: The `Style/TrailingComma` cop no longer exists. Please use `Style/TrailingCommaInLiteral` and/or `Style/TrailingCommaInArguments` instead.
(obsolete configuration found in /Users/anderson/puppet/src/puppet/bartwithrubocop/.rubocop.yml, please update it)
RuboCop failed!
```

Would you like me to post this on the original and have him do it, or do you want to fix it up?
(On fixing this, it works correctly.)

**Comment by MikaelSmith** [comment]
I'll fix it up.

**Comment by MikaelSmith** [comment]
Went with disabling both Style/TrailingCommaInLiteral and Style/TrailingCommaInArguments /cc @ghoneycutt

**Comment by andersonmills** [comment]
Notably, we now receive rubocop style violations on the module skeleton. :) That seems like a good thing to fix immediately if we're actually adding this.

```
$ bundle exec rake rubocop
Running RuboCop...
Inspecting 4 files
C.C.

Offenses:

Gemfile:3:46: C: Style/UnneededInterpolation: Prefer to_s over string interpolation.
puppetversion = ENV.key?('PUPPET_VERSION') ? "#{ENV['PUPPET_VERSION']}" : ['>= 3.3']
                                             ^^^^^^^^^^^^^^^^^^^^^^^^^^
spec/classes/init_spec.rb:3:1: C: Style/EmptyLinesAroundBlockBody: Extra empty line detected at block body beginning.

4 files inspected, 2 offenses detected
RuboCop failed!
```

**Comment by MikaelSmith** [comment]
Fixed up.

**Comment by andersonmills** [comment]
Fantastico.  :+1:

File: lib/puppet/module_tool/skeleton/templates/generator/Rakefile [hunk 1]
```diff
@@ -1,17 +1,32 @@
...
+    sh "ruby -c #{ruby_file}" unless ruby_file =~ %r{spec/fixtures}
   end
   Dir['templates/**/*.erb'].each do |template|
     sh "erb -P -x -T '-' #{template} | ruby -c"
   end
 end
+
+desc 'Run metadata_lint, lint, validate, and spec tests.'
+task :test do
+  [:metadata_lint, :lint, :validate, :validate_readme, :spec].each do |test|
```
Comment (ghoneycutt) [inline lib/puppet/module_tool/skeleton/templates/generator/Rakefile]: I coped this from a Rakefile that had `:validate_readme`. It is a total one off and should not be here.

Comment (ghoneycutt) [inline lib/puppet/module_tool/skeleton/templates/generator/Rakefile]: Fixed in PR #4946

**Pull request merged**